edb/edgeql/compiler/viewgen.py

```diff
@@ -446,6 +473,133 @@ def gen_pointers_from_defaults(
     return [result[i] for i in topological.sort(graph, allow_unresolved=True)]
 
 
+def _apply_rewrites(
+    pointers: Dict[s_pointers.Pointer, Optional[irast.Set]],
+    specified_ptrs: Set[sn.UnqualName],
+    rewrite_kind: qltypes.RewriteKind,
+    view_scls: s_objtypes.ObjectType,
+    ir_set: irast.Set,
+    stype: s_objtypes.ObjectType,
+    path_id_namespace: Optional[irast.Namespace],
+    view_rptr: Optional[context.ViewRPtr],
+    exprtype: s_types.ExprType,
+    ctx: context.ContextLevel,
+) -> List[Tuple[s_pointers.Pointer, irast.Set | None]]:
+    result: List[Tuple[s_pointers.Pointer, irast.Set | None]] = []
+
+    path_id = ir_set.path_id
+    scls_pointers = stype.get_pointers(ctx.env.schema)
+
+    # init set that will be used for __subject__
+    subject_path = irast.PathId.from_type(
+        ctx.env.schema,
+        stype,
+        typename=sn.QualName(module="__derived__", name="__subject__"),
+    )
+    subject_set = irast.Set(
+        path_id=subject_path,
+        typeref=subject_path.target,
+        shape=tuple(
+            [
+                (ptr_set, qlast.ShapeOp.ASSIGN)
+                for ptr_set in pointers.values()
+                if ptr_set
+            ]
+        ),
+    )
```

Comment:
@msullivan Can you take a look at this?

Is it ok to use stype as the type? I have a problem with `__subject__.title` compiling to a read from the table instead of the constant value that is provided in `subject_set.shape`.

Reply:
Yes, I think using `stype` is probably right. You'll need to do some plumbing on the pgsql side to get `__subject__` referring to the right stuff

Reply:
Hm... There's a question of implementation strategies, though:
The thing I had in mind was that you'd add some additional IR fields to represent the rewrites.
I haven't thought about what would take to make it work if you are trying to do it by rewriting the shape.

I think that you can probably make the pure shape approach work for `INSERT` but I don't know if it can be made to work for `UPDATE`, where subtypes may have rewrites that don't appear in the type being `UPDATE`d